tests/test_server.py

```diff
@@ -59,17 +59,125 @@ def udev_rules(self):
+                        0xc0,        # .End Collection                      54
+                    ]
+                },
+            }
         }
+        if 'profiles' in spec:
+            data['specification']['profiles'] = spec['profiles']
+        data['specification'].update(spec)
+
         response = client.post('/devices/add', json=data)
         assert response.status_code == 201
         return response.json()['id']
 
     def test_add_device(self, client, name='steelseries-rival310'):
         data = {
-            'shortname': name
+            'driver': 'steelseries',
+            'specification': {
+                'bus_info': [0x3, 0x01038, 0x1720],
+                'name': 'Steelseries Rival 310',
+                'shortname': 'steelseries-rival310',
+                'report_descriptors': {
+                    0: [
+                        0x05, 0x01,
+                        0x09, 0x02,
+                        0xa1, 0x01,
+                        0x09, 0x01,
+                        0xa1, 0x00,
+                        0xa1, 0x02,
+                        0x05, 0x09,
+                        0x19, 0x01,
+                        0x29, 0x08,
+                        0x15, 0x00,
+                        0x25, 0x01,
+                        0x95, 0x08,
+                        0x75, 0x01,
+                        0x81, 0x02,
+                        0x05, 0x01,
+                        0x09, 0x30,
+                        0x09, 0x31,
+                        0x16, 0x01, 0x80,
+                        0x26, 0xff, 0x7f,
+                        0x75, 0x10,
+                        0x95, 0x02,
+                        0x81, 0x06,
+                        0x09, 0x38,
+                        0x15, 0x81,
+                        0x25, 0x7f,
+                        0x75, 0x08,
+                        0x95, 0x01,
+                        0x81, 0x06,
+                        0xc0,
+                        0xa1, 0x02,
+                        0x05, 0x0c,
+                        0x0a, 0x38, 0x02,
+                        0x15, 0x81,
+                        0x25, 0x7f,
+                        0x75, 0x08,
+                        0x95, 0x01,
+                        0x81, 0x06,
+                        0xc0,
+                        0xa1, 0x02,
+                        0x06, 0xc1, 0xff,
+                        0x15, 0x00,
+                        0x26, 0xff, 0x00,
+                        0x75, 0x08,
+                        0x09, 0xf0,
+                        0x95, 0x02,
+                        0x81, 0x02,
+                        0xc0,
+                        0xc0,
+                        0xc0,
+                    ]
+                },
+                'profiles': [
+                    {
+                        'is_active': True,
+                    }
+                ],
+                'steelseries': {
+                    'version': 2
+                }
+            }
```

Comment:
Same here, why would a client need to know which specific report descriptor a Steelseries mouse has?